hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/SimpleAvroKeyGenerator.java

```diff
@@ -30,19 +33,36 @@
 
   public SimpleAvroKeyGenerator(TypedProperties props) {
     this(props, props.getString(KeyGeneratorOptions.RECORDKEY_FIELD_OPT_KEY.key()),
-        props.getString(KeyGeneratorOptions.PARTITIONPATH_FIELD_OPT_KEY.key()));
+        props.getString(KeyGeneratorOptions.PARTITIONPATH_FIELD_OPT_KEY.key()),
+        props.getString(KeyGeneratorOptions.INDEXKEY_FILED_OPT.key(),
+            KeyGeneratorOptions.INDEXKEY_FILED_OPT.defaultValue()));
   }
 
   SimpleAvroKeyGenerator(TypedProperties props, String partitionPathField) {
-    this(props, null, partitionPathField);
+    this(props, null, partitionPathField, null);
   }
 
   SimpleAvroKeyGenerator(TypedProperties props, String recordKeyField, String partitionPathField) {
+    this(props, recordKeyField, partitionPathField, null);
+  }
+
+  SimpleAvroKeyGenerator(TypedProperties props, String recordKeyField, String partitionPathField,
+      String indexKeyField) {
     super(props);
     this.recordKeyFields = recordKeyField == null
         ? Collections.emptyList()
         : Collections.singletonList(recordKeyField);
     this.partitionPathFields = Collections.singletonList(partitionPathField);
+    if (!StringUtils.isNullOrEmpty(indexKeyField) && !indexKeyField.equals(recordKeyField)) {
```

Comment:
`indexKeyField` must be equals to `recordKeyField` but not subset?

Reply:
Incorrect check here. But for the bucket index, indexKeyField can be the subset. There is an one2one match between bucketId and file groupId. Therefore, the record indexed by `colA` is always clustered to the same bucket and updated by key `colA` and `colB` with the old one stored in the bucket